### PR TITLE
[Fix] [Perf] Tune DSV4 EP48 stage2 for RCCL routing

### PR DESCRIPTION
**Pull request by yhl-amd**
## Summary

Fix the large-token DSV4 EP48/top-k=6 sorting-layout mismatch from #5459 and
select stage2 kernels using the actual ATOM RCCL routing distribution.

The tuned rows use `block_m=128`, so `moe_sorting` emits token routes in
128-row blocks. The original 32768/131072 stage2 names omitted `_sbm128`, which
made stage2 interpret the routes with a 64-row layout.

This PR now:

- adds `_sbm128` to the 32768- and 131072-token stage2 configs;
- rejects a FlyDSL stage2 kernel at runtime when its effective sort block does
  not match `moe_sorting`'s block size;
- validates the sort-block contract across all tuned FMoE CSV files;
- selects `t64x256x256_atomic_sbm128` for the 16384 and 32768 DSV4 EP48/top-k=6
  buckets after RCCL-aware microbenchmarking.

The 131072 bucket remains on `t64x128x256_atomic_sbm128` and retains the
sorting-layout correction.

## Why the RCCL-aware result differs from the dense tuner

The ordinary local-expert tuner routes all six choices to the 48 local experts.
ATOM's RCCL static decode instead all-gathers global top-6 routes selected from
384 experts and uses `expert_mask` to retain this rank's 48 experts. That leaves
about 0.75 valid local routes per input row, so atomic collisions are low and
route-level reduce output is unnecessary overhead.

The `_persist` suffix is also not an effective persistent schedule for this
FP8-activation stage2: `flydsl_moe_stage2` forces `a_dtype == "fp8"` onto the
non-persistent grid path.

## RCCL-aware MI355X results

Shape: gfx950, hidden=7168, inter=3072, 384 global / 48 local experts, top-k=6,
A8W4 per-1x32, Silu, SBM128. Each candidate ran sequentially on the same GPU
with three fixed seeds; each sample used 31 timed iterations. Times are medians.

| M | local routes | selected stage2 | prior/proposed stage2 | selected fused-MoE | prior/proposed fused-MoE |
|---:|---:|---:|---:|---:|---:|
| 13,000 | 9,933 | 385.9 us | 1,838.6 us (`t64x256 atomic_xcd4_persist`) | 945.5 us | 2,441.2 us |
| 16,384 | 12,466 | 443.0 us | 2,339.3 us (`t64x256 atomic_xcd4_persist`) | 1,150.3 us | 3,106.6 us |
| 32,768 | 24,928 | 758.5 us | 882.8 us (`t64x128 atomic`) | 2,044.9 us | 2,154.6 us |

At 32768, the selected kernel lowers isolated stage2 latency by 14.1% and full
fused-MoE latency by 5.1% versus the previous PR config.

Non-atomic variants were also tested. `t64x256 reduce_sbm128` measured
598.1/714.0/1323.2 us at M=13000/16384/32768, respectively, and lost to plain
atomic under RCCL routing.

## Validation

- `python3 -m pytest -q op_tests/tuning_tests/test_csv_validation.py`
  - 15 passed, 37 subtests passed
- `git diff --check`
- exact config lookup with EP sentinel normalization:
  - M=16384 -> `t64x256x256_atomic_sbm128`, block_m=128
  - M=32768 -> `t64x256x256_atomic_sbm128`, block_m=128
- direct DSV4 FMoE numerical checks using the edited model CSV:
  - M=16384: logits diff 0.000139483
  - M=32768: logits diff 0.000139857
- RCCL-aware reference checks for all benchmarked candidates:
  - maximum logits diff 0.001075 (threshold 0.01)
  - no NaN, OOM, invalid access, or traceback
- prior end-to-end validation retained from the sorting-layout fix:
  - 8x MI355X ATOM/RCCL GSM8K: 1319/1319 HTTP 200
  - flexible exact match 0.9469; strict exact match 0.9484

The CSV timing columns are retained from their original tuner rows; runtime
selection uses the key and kernel names, not those informational measurements.

### CODE REVIEW

File: aiter/configs/model_configs/dsv4_fp8fp4_tuned_fmoe.csv [hunk 1]
```diff
@@ -102,8 +102,8 @@ gfx950,256,2048,7168,3072,48,6,ActivationType.Silu,torch.bfloat16,torch.float8_e
...
-gfx950,256,32768,7168,3072,48,6,ActivationType.Silu,torch.bfloat16,torch.float8_e4m3fn,torch.float4_e2m1fn_x2,QuantType.per_1x32,1,0,128,0,6313.0874,flydsl_moe1_afp8_wfp4_bf16_t128x256x256_w2_bnt0_gui_fp8,0.0%,4525.02,flydsl_moe2_afp8_wfp4_bf16_t64x128x256_atomic,0.0%,10838.1074,0,0,0,0,0,
-gfx950,256,131072,7168,3072,48,6,ActivationType.Silu,torch.bfloat16,torch.float8_e4m3fn,torch.float4_e2m1fn_x2,QuantType.per_1x32,1,0,128,0,24929.8224,flydsl_moe1_afp8_wfp4_bf16_t128x256x256_w2_bnt0_gui_fp8,0.0%,17837.2,flydsl_moe2_afp8_wfp4_bf16_t64x128x256_atomic,0.0%,42767.0224,0,0,0,0,0,
+gfx950,256,32768,7168,3072,48,6,ActivationType.Silu,torch.bfloat16,torch.float8_e4m3fn,torch.float4_e2m1fn_x2,QuantType.per_1x32,1,0,128,0,6313.0874,flydsl_moe1_afp8_wfp4_bf16_t128x256x256_w2_bnt0_gui_fp8,0.0%,4525.02,flydsl_moe2_afp8_wfp4_bf16_t64x128x256_atomic_sbm128,0.0%,10838.1074,0,0,0,0,0,
```
Comment (valarLip) [inline aiter/configs/model_configs/dsv4_fp8fp4_tuned_fmoe.csv]: use flydsl_moe2_afp8_wfp4_bf16_t64x256x256_atomic_xcd4_persist_sbm128

**Review by valarLip — APPROVED** [review]
